### DISCUSSION TITLE
Filter top resource if there are any rows not matching in the joined table

### DISCUSSION THREAD

**Discussion by nilsi** [supabase/supabase#19860] [Questions]
Hi,

I am building a search where users can filter based on availability (start and end dates), free text, and checkbox filters.

Everything works great beside the filter for availability. I want to do a top-level filter on the `listings` table based on the inner embedded table `listings_availability`. If **any** rows in `listings_availability` do not match my filter, I want to **exclude** the top-level resource `listing`. Currently, it only excludes the top-level resource if **none** matches the embedded resource.

Is this possible or do I have to do two queries? One for checking the availability against the `listings_availability` and another to filter based on text and checkboxes in the `listings` table. Or shall I move all this to a RPC call maybe?

```
export async function getSearchListings(searchParams: ReadonlyURLSearchParams): Promise<ListingData[]> {
  const supabase = createClient();

  const params = new URLSearchParams(searchParams);
  const searchTerm = params.get("location")
  const facilities = params.get("facilities")
  const startDate = params.get("checkIn")
  const endDate = params.get("checkOut")

  const query = supabase.from('listings')
    .select(`
      *,
      listings_availability!inner(
        date,
        price,
        status
      )
    `)
    .eq('published', true)

  if (facilities) {
    query.contains('facilities', JSON.stringify(facilities.split(',')))
  }

  if (searchTerm) {
    query.textSearch('county_location', searchTerm, {
      type: 'websearch',
      config: 'swedish',
    })
  }

  if (startDate) {
    query.gte('listings_availability.date', startDate)
  }

  if (endDate) {
    query.lte('listings_availability.date', endDate)
  }

  if (endDate && startDate) {
    query.eq('listings_availability.status', "Available")
  }


  const { data, error } = await query

  console.log(query)
  console.log(data)

  if (error) {
    console.log(error.message);
    return [];
  }

  return data || [];
}
```

**Comment by pranavbadami** [comment]
@nilsi did you ever figure this one out?

**Reply by GaryAustin1** [reply]
This https://supabase.com/blog/postgrest-11-prerelease#anti-joins ?

**Reply by pranavbadami** [reply]
Hey @GaryAustin1 , thank you for the response. I'm specifically trying to support an "ALL" values filter on a related table, where I return the top-level rows where ALL of the related values exist. Is that possible? 

e.g. If i'm filter for `apple AND oranges` and have the following rows:  
```
[{
    name: "salad1"
    ingredients: [
        {name: "apple"},
        {name: "orange"}
    ]
},
{
    name: "salad2"
    ingredients: [
        {name: "apple"},
        {name: "strawberry"}
    ]
}]
```

i only want to return "salad1". Is that possible? I'm not seeing an easy way to do that, I'm only finding ways to filter for "ANY" values are contained, which would return both records above.
